chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+        rf.can_delete = True
+        rf.has_ctone = True
+        rf.has_dtcs = False           # TODO
+        rf.has_dtcs_polarity = False  # TODO
+        rf.has_bank = False           # TODO
+        rf.has_bank_names = False     # TODO
+        rf.has_name = True
+        rf.has_settings = True
+        rf.has_tuning_step = False  # hide in GUI, manage by code
+        # rf.valid_tuning_steps = list(STEPS)
+        # Attenuator = True
+        return rf
+
+    def process_mmap(self):
+        self._memobj = bitwise.parse(mem_format, self._mmap)
+
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number])
+
+    def get_memory(self, number):
+        _mem = self._memobj.memory[number]
+        _flag = self._memobj.flags[number]
+
+        mem = chirp_common.Memory()
+        mem.number = number
+
+        if _flag.hide_channel == 1:
+            mem.empty = True
+            return mem
+        mem.empty = False
+
+        if _mem.freq_flags == 0:
+            mem.freq = 5000 * (_mem.freq2 * 256 * 256 +
+                               _mem.freq1 * 256 +
+                               int(_mem.freq0))
+            mem.offset = 5000 * (_mem.offset_h * 256 + _mem.offset_l)
+        elif _mem.freq_flags == 20:
+            mem.freq = 6250 * (_mem.freq2 * 256 * 256 +
+                               _mem.freq1 * 256 +
+                               int(_mem.freq0))
+            mem.offset = 6250 * ((_mem.offset_h * 256) + _mem.offset_l)
+        elif _mem.freq_flags == 40:
+            mem.freq = 8333.3333 * (_mem.freq2 * 256 * 256 +
+                                    _mem.freq1 * 256 +
+                                    int(_mem.freq0))
+            mem.offset = 8333.3333 * ((_mem.offset_h * 256) + _mem.offset_l)
+        elif _mem.freq_flags == 60:
+            mem.freq = 9000 * (_mem.freq2 * 256 * 256 +
+                               _mem.freq1 * 256 +
+                               int(_mem.freq0))
+            mem.offset = 9000 * ((_mem.offset_h * 256) + _mem.offset_l)
+        else:
+            print(f"Unknown freq multiplier: {_mem.freq_flags}")
+            mem.freq = 1234567890
+            mem.offset = 0
+
+        # mem.tuning_step = STEPS[_mem.tuning_step]
+        mem.duplex = DUPLEX_DIRS[_mem.duplex]
+
+        mem.ctone = TONES[_mem.ctone]
+        mem.tmode = TONE_MODES[_mem.tmode]
+
+        mem.mode = MODES[_mem.mode]
+
+        # memory scan skip
+        if _flag.skip == 0:
+            mem.skip = ""      # None
+        elif _flag.skip == 1:
+            mem.skip = "S"     # memscan skip (aka "Skip")
+        elif _flag.skip == 3:
+            mem.skip = "P"     # 3 = mem&vfo ("Pskip")
+
+        # Channel names are encoded in 6x 6-bit groups spanning 5 bytes.
+        # Mask only the needed bits and then lookup character for each group
+        # Mem format: 0000AAAA AABBBBBB CCCCCCDD DDDDEEEE EEFFFFFF
+        mem.name = CODED_CHRS[((_mem.name0 & 0x0F) << 2)
```

Comment:
Please use `icf.warp_byte_size()` which handles this in both directions for you instead of all the bit math.

Reply:
Not working for me. Will debug locally and hopefully get it into a future PR.

Reply:
IC-R6 packs 6x 6-bit characters into 40 bits, the first 4 bits are unused/padding
Looking at the id31 and id5100.py drivers which icf.warp_byte_size() they seem to have 8x 7-bit characters packed into 7 bytes so no need for padding. Think those leading 4 bits is breaking it.

Reply:
This should let you pad/skip the bits: https://github.com/kk7ds/chirp/pull/1084

Reply:
Worked a treat, driver updated to use it both ways